images/virtualization-artifact/pkg/controller/vd/internal/attachee.go

```diff
@@ -52,8 +71,32 @@ func (h AttacheeHandler) Handle(ctx context.Context, vd *virtv2.VirtualDisk) (re
 		vd.Status.AttachedToVirtualMachines = append(vd.Status.AttachedToVirtualMachines, virtv2.AttachedVirtualMachine{
 			Name: vm.GetName(),
 		})
+
+		runningCondition, _ := service.GetCondition(vmcondition.TypeRunning.String(), vm.Status.Conditions)
+		if runningCondition.Status != metav1.ConditionFalse || vm.Status.Phase == virtv2.MachineStarting {
```

Comment:
Looks like you don't want to lock vd if vm Running cond is Unknown. But here you do

Reply:
Fixed.